cityofaustin/form-service-api
# Epic 1816: share media

QA Plan here:
https://github.com/cityofaustin/techstack/issues/1816

Short Version:
- Ref: https://github.com/cityofaustin/techstack/issues/1766
   - share evidence should now be /share-media, English + Spanish
- Ref: https://github.com/cityofaustin/techstack/issues/1782
  - ty confirmation email should no longer have ticket or witness info
- Ref: https://github.com/cityofaustin/techstack/issues/1763
  - fix email template for ticket info English + Spanish
  - test in outlook + non-outlook

---

Changelog:
- We now handle errors in the email sending code. Previously, exceptions were squashed.
- We had 2 variables in emailConfig ("recipiant" and "recipient") that are now consolidated into "recipient".
- We now only send an email to a user if the user's email exists.
- eliminate the word "evidence" from our codebase

Questions for @sergiogcx:
1. I changed some field names in the knack schema to purge the word "evidence" from our code. Is there some step I need to do to migrate knack?
2. During the email build step I run into `translate() Section: 'complaint', Key: '', Language: 'en' -- NOT FOUND` 5 times in the logs. This also happened on master so I guess it's not a big deal. And the template is filled out correctly. I dug into the code but could not understand the root cause. Just wanted to let you know.

We have no PR builds for the form-service-api. So we'll have to merge the form-service-api PR into master and the coa-forms PR into master in order to test. I have confidence in the code, but I definitely want @sergiogcx to look at the code first because there are some significant changes.
